### ISSUE TITLE
Higher order and mixed STFT phase derivatives

### ISSUE THREAD

**Issue by susnak** [ltfat/ltfat#37]
Implement second order and mixed phase derivatives of DGT as described in
http://perso.ens-lyon.fr/patrick.flandrin/IeeeSpl2011_20120306.pdf
The tricks are similar to the ones used for the first order derivative case.

**Comment by nholighaus** [comment]
In that case it should also be available for FBs as in the current filterbank reassignment

Von meinem Xperia™-Smartphone  gesendet

**Comment by susnak** [comment]
This could be done as  an additional flag to gabphasegrad (and filterbankphasegrad):
e.g.:
```
gabphasegrad(...,'22')
```
for 2nd order derivative in both directions, '21' and '12' for mixed and '11' would be the default.
I dont expect we need more than 9th order derivative :).
Are there any better propositions?

**Comment by jvnv** [comment]
I think it will be a valuable addition! Also, I think that extending the existing routines with an additional flag is the most appropriate way to integrate it. However, maybe the flags, '11', '21', etc., are a bit confusing since they could mean several things. Maybe one flag should be used for the order of the derivatives. Then, if the order isn't equal for both directions, use even one more flag to distinguish both directions. So, something like:

gabphasegrad(...,'1'),

if it is the first order in both directions, and 

gabphasegrad(...,'1', '2'), 

if it is the first order in one direction and the second order in the other direction. In this case,

gabphasegrad(...,'1')

is a short-hand for

gabphasegrad(...,'1', '1').

**Comment by susnak** [comment]
I am afraid that we cannot (easily!) use two flag groups with exactly the same flags. It would be possible to do e.g. 
gabphasegrad(...,'t1', 'f2')
or e.g.
gabphasegrad(...,{'1', '2'})
but that is not as pretty as 
gabphasegrad(...,'12')

**Comment by jvnv** [comment]
I see the problem regarding my suggestions. However, I also like what you proposed:

>> gabphasegrad(...,'t1', 'f2')

In my opinion, the meaning of this is very clear.

**Comment by susnak** [comment]
After spending some time on phase derivatives I have found out the following:

* Phase sucks!

* Phase derivatives depend on the phase convention.

* There are 3 phase conventions + at least three versions of formulas for STFT

* ```[tgrad,fgrad]=gabphasegrad(...)``` computes:
tgrad - time derivative of freq. invarinant phase
fgrad - minus frequency derivative of time invariant phase.

I tried to sumarize everything in ltfatnote042.

New functions related to this issue:

*```phased=gabphasederiv(dflag,...,phaseconv)``` where ```dflag``` can be 't', 'f', 'tt', 'ff' or 'tf'  and ```phaseconv``` can be 'freqinv', 'timeinv', 'symphase' or 'relative'

*```gabreassignadjust(...,mu)``` where mu is a parameter controlling how strongly the reassignment is done.
